count/Feature/TopicsFeature.swift

```diff
@@ -19,8 +19,7 @@ extension TopicCategory {
 
 struct TopicsFeature: Reducer {
     struct State: Equatable {
-        @PresentationState var quiz: ListeningQuizNavigationFeature.State?
-        @PresentationState var about: AboutFeature.State?
+        @PresentationState var destination: Destination.State?
```

Comment:
The `PreSettingsView` should always be visible as a detents sheet. However, SwiftUI breaks if you try to present another sheet/fullScreenCover while another sheet is presented. Therefore, in the reducer below, we have to jump through some hoops to ensure that the `PreSettingsView` is dismissed properly before `AboutView` or `QuizView` is presented, and then present it again once `TopicView` reappears.